### PR TITLE
Mobility

### PR DESCRIPTION
**Pull request by ArtemFediai**
I have addressed you concerns from the previous PR. Please take a look

### CODE REVIEW

File: docs/source/science/calculators/mobility/output_molecule.mol2
Comment (tneumann2021) [inline docs/source/science/calculators/mobility/output_molecule.mol2:1]: Should this not be in static?

Comment (tneumann2021) [reply docs/source/science/calculators/mobility/output_molecule.mol2:1]: or delete...

File: docs/source/science/calculators/mobility.rst [hunk 1]
```diff
@@ -0,0 +1,563 @@
+.. _science_calculators_mobility:
+
+Charge Carrier Mobility. Calculator **mobility**.
+=================================================
+
+
+The Nanomatch Mobility Workflow (Calculator "**mobility**") is a multiscale simulations workflow designed to calculate the hole mobility of organic semiconductors starting from the first-principles.
```
Comment (tneumann2021) [inline docs/source/science/calculators/mobility.rst]: hole mobility -> charge carrier mobility

File: docs/source/science/calculators/mobility.rst [hunk 2]
```diff
@@ -0,0 +1,563 @@
...
+Users only need to provide the molecular structure as an InChI key. The workflow then takes over, progressing from the atomic properties to the macroscopic hole mobility, ensuring an efficient and comprehensive computational protocol.
```
Comment (tneumann2021) [inline docs/source/science/calculators/mobility.rst]: This is correct technically, but for DiaDEM users only click a button.
Suggestion: "The workflow progresses from the single molecular structure to the macroscopic...

File: docs/source/science/calculators/mobility.rst [hunk 3]
```diff
@@ -0,0 +1,563 @@
...
+   * - **Nanomatch Software**
+     - **Scientific Role**
+     - **Illustration**
+   * - `Parametrizer <http://docs.nanomatch.de/nanomatch-modules/Parametrizer/Parametrizer.html>`_
+     - | Geometry optimization
+     - .. image:: mobility/parametrizer.png
+          :width: 300px
+          :align: center
+   * - `DihedralParametrizer <http://docs.nanomatch.de/nanomatch-modules/DihedralParametrizer/DihedralParametrizer.html>`_
+     - | Parametrization of the total energy
```
Comment (tneumann2021) [inline docs/source/science/calculators/mobility.rst]: Computation of intramolecular forcefields

File: docs/source/science/calculators/mobility.rst [hunk 4]
```diff
@@ -0,0 +1,563 @@
...
+       | depending on dihedral angles
+     - .. image:: mobility/dhp.png
+          :width: 300px
+          :align: center
+   * - `Deposit <http://docs.nanomatch.de/nanomatch-modules/Deposit/Deposit.html>`_
+     - | Simulation of the physical deposition
```
Comment (tneumann2021) [inline docs/source/science/calculators/mobility.rst]: physical vapor deposition (PVD)

File: docs/source/science/calculators/mobility.rst [hunk 5]
```diff
@@ -0,0 +1,563 @@
...
+
+Molecular Structure Optimization
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+*obabel*, *xtb*, and *Density Functional Theory (DFT)* are used to generate initial 3D conformers, pre-optimize and optimize the geometry, and compute partial charges via an electrostatic potential (ESP) fit for single molecules in a vacuum. DFT optimization is performed with def2-TZVP/B3LYP level of theory.
+
+Morphology Generation
+~~~~~~~~~~~~~~~~~~~~~
+
+The *DEPOSIT protocol* [1]_ simulates physical vapor deposition to generate thin-film morphologies with atomistic resolution. This involves Monte Carlo (MC) based basin hopping with simulated annealing (SA) to model intermolecular interactions during deposition. In total, 1000 molecules are deposited into a box with a base size of 100x100 Å. After this, the top and bottom 7 Å are cut out, and periodic copies are added in the x and y axes to increase its base size to 300x300 Å.
```
Comment (tneumann2021) [inline docs/source/science/calculators/mobility.rst:65]: If we do only 1000 molecules, 100x100A may lead to flat morphologies, especially for smaller molecules. Do 90x90 and 1500 molecules ?  Needs to be benchmarked...

File: docs/source/science/calculators/mobility.rst [hunk 6]
```diff
@@ -0,0 +1,563 @@
...
+
+Electronic Structure Calculation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Using the *QuantumPatch method* [2]_, energy disorder, electronic couplings, and reorganization energies are calculated by self-consistently equilibrating the charge densities of a subset of molecules in their unique environments.
```
Comment (tneumann2021) [inline docs/source/science/calculators/mobility.rst:70]: We do not compute reorganization energies in this workflow, do we?

File: docs/source/science/calculators/mobility.rst [hunk 7]
```diff
@@ -0,0 +1,563 @@
...
+Zero-field mobility is extrapolated to the zero-field limit assuming Poole-Frenkel field dependence.
+
+Parameters of the kMC simulations:
+
+- **Fields**: three fields are applied: 0.02 0.03 0.04 eV/nm.
+- **Morphology** and **replicas**: for every field value, 10 independent morphologies are generated using the stochastic expansion scheme, including HOMO/LUMO/Js distributions derived from the *QuantumPatch* simulations.
+- **Temperature** is 300 K.
+- **Convergence criterion**: either the fluctuation parameter "iv_fluctuation" below 0.05, or "max_iterations" exceeds 5E6.
+- **Number of Charge Carriers**: 30.
+- **Neighbours**: 120, meaning a charge carrier can jump to the nearest 120 neighboring molecules.
```
Comment (tneumann2021) [inline docs/source/science/calculators/mobility.rst]: I would not necessarily mention this. This is very detailed information.

File: docs/source/science/calculators/mobility.rst [hunk 8]
```diff
@@ -0,0 +1,563 @@
...
+- **First shell**: Self-consistent DFT shell def2-SVP/BP86, radius 15 Å
+- **Second shell**: Self-consistent DFTB, radius 25 Å
+- **Third shell**: Static DFTB, radius 60 Å
+
+
+Structure Expansion
+~~~~~~~~~~~~~~~~~~~
+
+To bridge the scales from atomistic resolution to the device level, a stochastic expansion scheme *EDCM* is used to expand the thin-film morphologies,
+drawing electronic couplings and site energies from distributions analyzed in the QuantumPatch method.
```
Comment (tneumann2021) [inline docs/source/science/calculators/mobility.rst]: How large is the sample we then simulate?

File: docs/source/science/calculators/mobility.rst [hunk 9]
```diff
@@ -0,0 +1,563 @@
...
+The workflow uses the LightForge package to simulate field-dependent mobility, taking into account percolation and many-body effects [4]_.
+Zero-field mobility is extrapolated to the zero-field limit assuming Poole-Frenkel field dependence.
+
+Parameters of the kMC simulations:
+
+- **Fields**: three fields are applied: 0.02 0.03 0.04 eV/nm.
+- **Morphology** and **replicas**: for every field value, 10 independent morphologies are generated using the stochastic expansion scheme, including HOMO/LUMO/Js distributions derived from the *QuantumPatch* simulations.
+- **Temperature** is 300 K.
+- **Convergence criterion**: either the fluctuation parameter "iv_fluctuation" below 0.05, or "max_iterations" exceeds 5E6.
+- **Number of Charge Carriers**: 30.
```
Comment (tneumann2021) [inline docs/source/science/calculators/mobility.rst]: Add: In the expanded simulation box of axbxc nm3 this results in a charge carrier concentration of...

File: docs/source/science/calculators/mobility.rst [hunk 10]
```diff
@@ -0,0 +1,563 @@
...
+- **Neighbours**: 120, meaning a charge carrier can jump to the nearest 120 neighboring molecules.
+
+Output
+------
+
+Parsed Output
+~~~~~~~~~~~~~
+The data below will be displayed as the workflow ends (backend name: `result.yml`):
```
Comment (tneumann2021) [inline docs/source/science/calculators/mobility.rst:124]: what is a backend name? The user will only see the frontend ;-) But I get the point: We should fix this description once we really no how this results file will be made available in the frontend

File: docs/source/science/calculators/mobility.rst [hunk 11]
```diff
@@ -0,0 +1,563 @@
...
+* **PK**: P. Kordt et al. (2015), "Modeling of Organic Light Emitting Diodes: From Molecular to Device Properties," *Adv. Funct. Mater.*, 25, 1955-1971. URL: https://doi.org/10.1002/adfm.201403004
+* **AM**: A. Massé et al. (2016), "Ab initio charge-carrier mobility model for amorphous molecular semiconductors," *Phys. Rev. B*, 93, 195209. URL: https://doi.org/10.1103/PhysRevB.93.195209
+* **DE**: D. Evans et al. (2016), "Estimation of charge carrier mobility in amorphous organic materials using percolation corrected random-walk model," *Org. Electron.*, 29, 50–56. URL: https://doi.org/10.1016/j.orgel.2015.11.021
+* **PF**: P. Friederich et al. (2016), "Molecular Origin of the Charge Carrier Mobility in Small Molecule Organic Semiconductors," *Adv. Funct. Mater.*, 26, 5757–5763. URL: https://doi.org/10.1002/adfm.201601807
+* **NK**: N. Kotadiya et al. (2018), "Rigorous Characterization and Predictive Modeling of Hole Transport in Amorphous Organic Semiconductors," *Adv. Electron. Mater.*, 4, 1800366. URL: https://doi.org/10.1002/aelm.201800366
+
+
+Reference
```
Comment (tneumann2021) [inline docs/source/science/calculators/mobility.rst]: References

File: docs/source/science/calculators/mobility.rst [hunk 12]
```diff
@@ -0,0 +1,563 @@
...
+- **Fields**: three fields are applied: 0.02 0.03 0.04 eV/nm.
+- **Morphology** and **replicas**: for every field value, 10 independent morphologies are generated using the stochastic expansion scheme, including HOMO/LUMO/Js distributions derived from the *QuantumPatch* simulations.
+- **Temperature** is 300 K.
+- **Convergence criterion**: either the fluctuation parameter "iv_fluctuation" below 0.05, or "max_iterations" exceeds 5E6.
+- **Number of Charge Carriers**: 30.
+- **Neighbours**: 120, meaning a charge carrier can jump to the nearest 120 neighboring molecules.
+
+Output
+------
+
```
Comment (tneumann2021) [inline docs/source/science/calculators/mobility.rst:113]: Before the detailed output in the results dictionary, we could list all the Properties (in terms of DiaDEM Properties as available in the Properties-section of the docs) that are provided by this calculator: 

This Calculator provides the following Properties: 
 - ... (and include a link to the page in the Properties-section)

File: docs/source/science/properties/mobility.rst [hunk 13]
```diff
@@ -0,0 +1,103 @@
+.. _science_properties_mobility:
+
+Charge Carrier Mobility
```
Comment (tneumann2021) [inline docs/source/science/properties/mobility.rst]: We will provide electron and hole mobility as two different "Properties", computed by the same calculator, right? So if the user initiates a computation for "Electron Mobility" choosing this property from a dropdown in the frontend, he should find the label "Electron Mobility" somewhere in the title to make the connection. Suggestion: "Electron and Hole Mobility" or "Charge carrier mobility: electron mobility and hole mobiliy". The more similar the strings, the more easy it is to connect between frontend and documentation.

**Review by tneumann2021 — CHANGES_REQUESTED** [review]

File: docs/source/science/calculators/mobility/output_molecule.mol2
Comment (ArtemFediai) [reply docs/source/science/calculators/mobility/output_molecule.mol2:1]: removed structure.cml and *mol2 as these are already in _static and used from there.

File: docs/source/science/calculators/mobility.rst (hunk 1 above)
Comment (ArtemFediai) [reply docs/source/science/calculators/mobility.rst]: done

File: docs/source/science/calculators/mobility.rst (hunk 2 above)
Comment (ArtemFediai) [reply docs/source/science/calculators/mobility.rst]: redone the sentence using suggestions

File: docs/source/science/calculators/mobility.rst (hunk 3 above)
Comment (ArtemFediai) [reply docs/source/science/calculators/mobility.rst]: changed to "Computation of intramolecular forcefields'"

File: docs/source/science/calculators/mobility.rst (hunk 4 above)
Comment (ArtemFediai) [reply docs/source/science/calculators/mobility.rst]: changed

File: docs/source/science/calculators/mobility.rst (hunk 5 above)
Comment (ArtemFediai) [reply docs/source/science/calculators/mobility.rst:65]: this is my current production settings. with 100x100 I mean Lx=Ly=50.

Comment (ArtemFediai) [reply docs/source/science/calculators/mobility.rst:65]: So, I did not change here anything, but we have to somehow control this point

File: docs/source/science/calculators/mobility.rst (hunk 6 above)
Comment (ArtemFediai) [reply docs/source/science/calculators/mobility.rst:70]: no. Assume that it is 200 meV for both e and h.

File: docs/source/science/calculators/mobility.rst (hunk 7 above)
Comment (ArtemFediai) [reply docs/source/science/calculators/mobility.rst]: removed

File: docs/source/science/calculators/mobility.rst (hunk 9 above)
Comment (ArtemFediai) [reply docs/source/science/calculators/mobility.rst]: added.

File: docs/source/science/calculators/mobility.rst (hunk 8 above)
Comment (ArtemFediai) [reply docs/source/science/calculators/mobility.rst]: 40 40 40 -- added

File: docs/source/science/calculators/mobility.rst (hunk 12 above)
Comment (ArtemFediai) [reply docs/source/science/calculators/mobility.rst:113]: done. Note: dipole is not yet added as the property, and no page exist for it here

File: docs/source/science/calculators/mobility.rst (hunk 10 above)
Comment (ArtemFediai) [reply docs/source/science/calculators/mobility.rst:124]: exactly, I can mark things with todo to traverse through docs before putting them to production ..

File: docs/source/science/calculators/mobility.rst (hunk 11 above)
Comment (ArtemFediai) [reply docs/source/science/calculators/mobility.rst]: done

File: docs/source/science/properties/mobility.rst (hunk 13 above)
Comment (ArtemFediai) [reply docs/source/science/properties/mobility.rst]: changed to Electron and Hole Mobility

**Review by tneumann2021 — APPROVED** [review]

**Pull request merged**